src/shared/serviceidresolver/serviceidresolver.go

```diff
@@ -29,17 +29,26 @@ func ResolvePodToServiceIdentityUsingAnnotationOnly(pod *corev1.Pod) (string, bo
 }
 
 // ResolvePodToServiceIdentity resolves a pod object to its otterize service ID, referenced in intents objects.
-// This is done by recursion over the pod's owner reference hierarchy until reaching a root owner reference.
+// It calls GetOwnerObject to recursively iterates over the pod's owner reference hierarchy until reaching a root owner reference.
```

Comment:
Oops, the changes in this file are not mine, they are here because I pulled from main 🤦